docs/TECHNICAL_DESIGN.md

```diff
@@ -0,0 +1,155 @@
+## Technical Design
+
+Carrot follows the Facebook Flux pattern:
+
+<img src="./flux-diagram-white-background.png" style="width: 100%;" />
+
+_Image source: the [Flux repo](https://github.com/facebook/flux)_
+
+It's important to note, Flux isn't a particular software library, instead it's an architectural pattern that's been implemented now in countless React apps and in many React web frameworks.
+
+The key idea in Flux is having a unidirectional data flow:
+
+<img src="./flux-simple-f8-diagram-with-client-action-1300w.png" style="width: 100%;" />
+
+_Image source: [Flux concepts](https://github.com/facebook/flux/tree/master/examples/flux-concepts)_
+
+Here is some background information on Flux:
+
+- https://github.com/facebook/flux - The Flux repo (documentation of the pattern, as well as a JavaScript Dispatcher implemntation)
+- https://github.com/facebook/flux/tree/master/examples/flux-concepts - High-level Flux concepts
+- https://facebook.github.io/flux/docs/in-depth-overview.html - Flux in depth
+- https://github.com/kgann/cljs-flux - An example implementation of the Flux pattern in ClojureScript
+
+### Action Creators
+
+An action creator is called by a user interaction or an API request finishing.
+
```

Comment:
I think a key piece of info here @thepug is what our action creators do. Flux leaves this very abstract, just that in the end they emit an action to the Dispatcher that has the info needed about what state changed, and that the action be semantic, not overly impl. specific.

But what's interesting to us is, what does it mean to us about what an action creator will do. The responsibilities I see are things like:

- Pass an action to the dispatcher
- Make API invocations
- Update cookies
- Update URL routes

Do each of these belong in an action creator? Is this exhaustive? Are there other responsibilities?